R/prob-roc_aunp.R

```diff
@@ -0,0 +1,160 @@
+#' Area under the ROC curve of each class against the rest, using the a priori
+#' class distribution
+#'
+#' `roc_aunp()` is a multiclass metric that computes the area under the ROC
+#' curve of each class against the rest, using the a priori class distribution.
+#' This is equivalent to `roc_auc(estimator = "macro_weighted")`.
+#'
+#' Like the other ROC AUC metrics, `roc_aunp()` defaults to allowing
+#' `pROC::roc()` control the direction of the computation, but allows you to
+#' control this by passing `options = list(direction = "<")` or any other
+#' allowed direction value. pROC advises setting the `direction` when doing
+#' resampling so that the AUC values are not biased upwards.
+#'
+#' Generally, an ROC AUC value is between `0.5` and `1`, with `1` being a
+#' perfect prediction model. If your value is between `0` and `0.5`, then
+#' this implies that you have meaningful information in your model, but it
+#' is being applied incorrectly because doing the opposite of what the model
+#' predicts would result in an AUC `>0.5`.
+#'
+#' @family class probability metrics
+#' @templateVar metric_fn roc_aunp
+#' @template return
+#' @template event_first
+#'
+#' @section Multiclass:
+#' This multiclass method for computing the area under the ROC curve uses the
+#' uniform class distribution and is equivalent to
```

Comment:
A priori class distribution? Not uniform right?

Reply:
Fixed in 1c52d911607b590aa7a1e433cc515160aff72925